### PR TITLE
Use dynamic golang version matrix

### PR DESCRIPTION
**Pull request by dopey**

### CODE REVIEW

File: .github/workflows/goTest.yml [hunk 1]
```diff
@@ -16,16 +20,35 @@ on:
...
+
   test:
+    needs: set-go-matrix
     runs-on: ubuntu-latest
     env:
       GOPRIVATE: go.step.sm
     strategy:
-      matrix:
-        go: ['1.18', '1.19']
+      matrix: ${{ fromJson(needs.set-go-matrix.outputs.matrix) }}
```
Comment (maraino) [inline .github/workflows/goTest.yml:48]: The value of matrix seems to be `{\"include\":[{\"go\":\"1.19\"}]}`, is that `include` ok?

File: .github/workflows/goBuild.yml [hunk 2]
```diff
@@ -4,20 +4,43 @@ on:
...
+
   build:
+    needs: set-go-matrix
     runs-on: ubuntu-latest
     env:
       GOPRIVATE: go.step.sm
     strategy:
-      matrix:
-        go: ['1.18', '1.19']
+      matrix: ${{ fromJson(needs.set-go-matrix.outputs.matrix) }}
```
Comment (maraino) [inline .github/workflows/goBuild.yml:43]: The value of matrix seems to be {\"include\":[{\"go\":\"1.19\"}]}, is that include ok?

File: .github/workflows/goTest.yml (hunk 1 above)
Comment (dopey) [reply .github/workflows/goTest.yml:48]: it is. I don't know why it's necessary but 🤷 . https://docs.github.com/en/actions/learn-github-actions/expressions#example-returning-a-json-object

File: .github/workflows/goBuild.yml (hunk 2 above)
Comment (dopey) [reply .github/workflows/goBuild.yml:43]: See above answer.

File: .github/workflows/goTest.yml (hunk 1 above)
Comment (maraino) [reply .github/workflows/goTest.yml:48]: Looks like by https://docs.github.com/en/actions/learn-github-actions/expressions#example-returning-a-json-object

**Review by maraino — APPROVED** [review]

**Pull request merged**